credentials/alts/alts.go

```diff
@@ -137,19 +137,23 @@ type altsTC struct {
 }
 
 // NewClientCreds constructs a client-side ALTS TransportCredentials object.
-func NewClientCreds(opts *ClientOptions) credentials.TransportCredentials {
+func NewClientCreds(opts *ClientOptions) (credentials.TransportCredentials, error) {
 	return newALTS(core.ClientSide, opts.TargetServiceAccounts, opts.HandshakerServiceAddress)
 }
 
 // NewServerCreds constructs a server-side ALTS TransportCredentials object.
-func NewServerCreds(opts *ServerOptions) credentials.TransportCredentials {
+func NewServerCreds(opts *ServerOptions) (credentials.TransportCredentials, error) {
 	return newALTS(core.ServerSide, nil, opts.HandshakerServiceAddress)
 }
 
-func newALTS(side core.Side, accounts []string, hsAddress string) credentials.TransportCredentials {
+func newALTS(side core.Side, accounts []string, hsAddress string) (credentials.TransportCredentials, error) {
+	var err error
 	once.Do(func() {
-		vmOnGCP = isRunningOnGCP()
+		vmOnGCP, err = isRunningOnGCP()
```

Comment:
Does it make more sense to run `isRunningOnGCP` from an `init` function?

Reply:
Also, since we are returning error from the `NewClientCreds` and `NewServerCreds` functions if the platform is not GCP, do we really need to check this `vmOnGCP` boolean from `ServerHandshake` and `ClientHandshake` methods?